### ISSUE TITLE
Product don't include anchor category in multistore Magento

### ISSUE THREAD

**Issue by gustadev84** [Smile-SA/elasticsuite#2218]
When I have a category only enabled in a single store of a multistore Magento, the product in the sub-category is not being listed on the anchor category.

### Preconditions

Magento Version : CE 2.3.5-p2
ElasticSuite Version : 2.9.6
Environment : Production Mode

### Steps to reproduce
Magento has five stores created.

A category and its subcategories are only enabled in one of the stores and disabled on Main Store. The anchor category is not listing the products of subcategories on the frontend.

### Expected result
List subcategories products on anchor category inside a not main store.

### Actual result
No products of subcategories listed on anchor category.

======

If I enable the subcategory on main store, the product is listed on the anchor category of the store view.

Thanks!

**Comment by vahonc** [comment]
Hello @gustadev84,

Are you sure that your issue is linked to the ES module, maybe it's just a wrong configuration of multistore websites or something was not indexed by ES indices?

Did you try to disable or remove temporary our module? If so, what's the result which you get?

BR,
Vadym

**Comment by gustadev84** [comment]
Hi @vahonc 

Thanks for your answer and sorry for the late reply.

I sorted the issue and it's not related to Smile Elasticsuite, but the module is impacted by the issue.

Elasticsuite loads data from catalog_category_product_index, right? So, when it's created, if the anchor category is not enabled on the main store, the child category won't be indexed together in a not main store. The issue is on `magento/module-catalog/Model/Indexer/Category/Product/AbstractAction.php::fillTempCategoryTreeIndex` function that's always using the $currentStoreId var as 0 (declared on this class).

I sorted it by patching three methods of `magento/module-catalog/Model/Indexer/Category/Product/AbstractAction.php ` class:

1. fillTempCategoryTreeIndex
- Added a new parameter storeId (default null).

2. makeTempCategoryTreeIndex
- Added a new parameter storeId (default null).
- Fill the storeId parameter on the call of fillTempCategoryTreeIndex.

3. createAnchorSelect
- Fill $store->getId() parameter on makeTempCategoryTreeIndex call.

Full reindex after that and it fixed the issue.

Thanks!

Gustavo
